libraries/AP_HAL_ChibiOS/hwdef/SkyRukh_Surge_H7/hwdef.dat

```diff
@@ -0,0 +1,202 @@
+# Hardware definition file for processing by chibios_pins.py
+# SkyRukh Surge H7 FC board with H743 MCU, BMI088, Barometer,
+# ICM20602, TF-card, FRAM, and OSD
+
+# MCU class and specific type
+MCU STM32H7 STM32H743xx
+
+# crystal frequency
+OSCILLATOR_HZ 16000000
+
+# board ID.
+APJ_BOARD_ID AP_HW_SkyRukh_Surge_H7
+
+# flash size
+FLASH_SIZE_KB 2048
+
+# ChibiOS system timer
+define STM32_ST_USE_TIMER 2
+
+# bootloader takes first sector
+FLASH_RESERVE_START_KB 128
+
+# I2C Buses
+
+# I2C2, GPS connector
+PB10 I2C2_SCL I2C2
+PB11 I2C2_SDA I2C2
+
+# I2C2, external
+PF14 I2C4_SCL I2C4
+PF15 I2C4_SDA I2C4
+
+# order of I2C buses
+I2C_ORDER I2C2 I2C4
+
+# order of UARTs (and USB)
+SERIAL_ORDER OTG1 USART2 USART6 USART3 UART7 UART4 UART5 UART8
+
+# USB
+PA11 OTG_FS_DM OTG1
+PA12 OTG_FS_DP OTG1
+
+# pins for SWD debugging
+PA13 JTMS-SWDIO SWD
+PA14 JTCK-SWCLK SWD
+
+# SBUS
+PD14 SBUS_INV OUTPUT HIGH
+
+# RC input
+PA10 TIM1_CH3 TIM1 RCININT PULLDOWN LOW
+
+# UARTs
+
+PD5     USART2_TX USART2
+PD6     USART2_RX USART2
+PD3	    USART2_CTS USART2
+PD4	    USART2_RTS USART2
+
+PD8	    USART3_TX USART3
+PD9	    USART3_RX USART3
+
+PD1     UART4_TX UART4 NODMA
+PD0     UART4_RX UART4 NODMA
+
+PB13	UART5_TX UART5
+PB12	UART5_RX UART5
+
+PG14    USART6_TX  USART6
+PG9     USART6_RX  USART6
+PG15	USART6_CTS USART6
+PG12	USART6_RTS USART6
+
+PE8     UART7_TX UART7 
+PF6     UART7_RX UART7 
+
+PE0     UART8_RX UART8
+PE1     UART8_TX UART8
+
+# drdy pins
+PB3 DRDY2_ICM20602 INPUT
+PE4 DRDY3_BMI088_GYRO1	INPUT
+PI8 DRDY3_BMI088_ACC1	INPUT
+PI7 DRDY6_BMI088_GYRO2	INPUT
+PF2 DRDY2_BMI088_ACC2	INPUT
+
+# ADC inputs
+define HAL_USE_ADC TRUE
+define STM32_ADC_USE_ADC1 TRUE
+
+# ADC pins
+PA0     BATT_VOLT_SENS ADC1      SCALE(1)
+PA1     BATT_CURR_SENS ADC1      SCALE(2)
+
+# Motors
```

Comment:
on an H7, you should be able to get at least four outputs bi-dir capable with some DMA adjustments, maybe none